R/admiral-package.R

```diff
@@ -21,7 +21,7 @@
 #'   str_trim str_to_lower str_to_title str_to_upper str_length str_locate
 #' @importFrom lubridate as_datetime ceiling_date date days duration floor_date is.Date is.instant
 #'             rollback time_length %--% ymd ymd_hms weeks years hours minutes
-#' @importFrom tidyr drop_na nest pivot_longer pivot_wider unnest
+#' @importFrom tidyr drop_na nest pivot_longer pivot_wider unnest crossing fill
```

Comment:
Would it be easier to maintain the package if we specify `@importFrom` in the roxygen headers of the functions for which the imports are required? That way, within any given {admiral} function, it is easier to see the packages from which imported functions originate--for example, if I see that `crossing()` is called from `derive_locf_records()`, then I only have to look in its roxygen header to see that `crossing()` is imported from {tidyr}; whereas if we specify all the imports within admiral-package.R, then it's not clear.

Reply:
I think for smaller packages it makes more sense to do it the way you are suggesting.  

However, for large packages, like admiral, I think it makes for more maintenance with filling out `@importFrom` in the roxygen headers.  If six admiral functions all need `crossing()` do you have to fill out each `@importFrom` in the roxygen headers?  I can't remember as its been a while since ive done it the other way, but think that is the case.  

This way we only have to fill it out one time for the six functions.

Reply:
I think you would ideally fill out `@importFrom` in the roxygen headers for all six functions, so that one is not dependent on another having `@importFrom` specified in its roxygen header, but personally I'd prefer to do that, _especially_ for a large package, because it would be harder to find a function within admiral-packages.R when there are lots of imported functions. I think @statasaurus once advised me to follow the approach of specifying all imported functions via `@importFrom` tags in the roxygen header.

Reply:
Well.. I think since this is an established practice on `admiral` so we should just continue as is.  I personally like it this way, but see the value of the individual calls to help new developers understand where everything resides.  Also, I always just hit `f1` to see where a function lives - perhaps we can mention this somewhere if folks are having trouble locating a function

Reply:
I think it is almost impossible to maintain `@importFrom` in the roxygen headers because if you would miss to add `crossing()` in some of the six files, the package would still work and all checks pass.